src/payment/data/sagas.js

```diff
@@ -10,34 +10,30 @@ import {
   submitPayment,
 } from './actions';
 
+// Sagas
+import { handleErrors, handleMessages } from '../../feedback';
+
 // Services
 import * as PaymentApiService from './service';
-
 import { checkout as checkoutCybersource } from '../payment-methods/cybersource';
 import { checkout as checkoutPaypal } from '../payment-methods/paypal';
 import { checkout as checkoutApplePay } from '../payment-methods/apple-pay';
-import { handleErrors, handleMessages } from '../../feedback';
 
 export const paymentMethods = {
   cybersource: checkoutCybersource,
   paypal: checkoutPaypal,
   'apple-pay': checkoutApplePay,
 };
 
-
-function* isBasketProcessing() {
-  return yield select(state => state.payment.basket.isBasketProcessing);
-}
-
-export function* handleFetchBasket() {
+/**
+ * Many of the handlers here have identical error handling, and are also all processing the same
+ * sort of response object (a basket).  This helper is just here to dedupe that code, since its
+ * all identical.
+ */
+export function* catchBasketError(handler) {
```

Comment:
I pulled this in from @MatthewPiatetsky 's PR (https://github.com/edx/frontend-app-payment/pull/67) and used it in all our handlers below.  Matthew, this may make your merge a bit interesting, but it also means it's already tested for you. :)